### PR TITLE
make ServiceRegistry typed

### PR DESCRIPTION
**Pull request by oalyman**
this PR makes the ServiceRegistry typed so that the type of the returned instances can be inferred by your IDE / editor.
Also I added the vars folder as a source directory.
Both changes make it easier navigating the code in your IDE / editor and remove a little bit of boilerplate (".class.name" usage).

### CODE REVIEW

**Review by clemensutschig — APPROVED** [review]

**Comment by metmajer** [comment]
@oalyman please give us the time to test your contributions internally before we accept them, and don't merge them too soon. Thanks!

**Comment by metmajer** [comment]
@oalyman next time when you do a merge, please squash your commits into a single commit for a clean git history

**Comment by metmajer** [comment]
The critical feedback above side, and most importantly, thanks for making this project a better one 👍

**Pull request merged**
